spec/controllers/vm_cloud_controller_spec.rb

```diff
@@ -56,6 +61,33 @@
       expect(response.body).to_not be_empty
     end
 
+    it 'can open instance resize tab' do
+      post :explorer
+      expect(response.status).to eq(200)
+
+      post :x_button, :params => {:pressed => 'vm_protect', :id => vm_openstack.id}
+      expect(response.status).to eq(200)
+
+      allow(controller).to receive(:x_node).and_return("v-#{vm_openstack.compressed_id}")
+
+      post :x_button, :params => {:pressed => 'instance_resize', :id => vm_openstack.id}
+      expect(response.status).to eq(200)
+      expect(response).to render_template(:partial => 'vm_common/_resize')
+    end
+
+    it 'can resize an instance' do
+      flavor = FactoryGirl.create(:flavor_openstack)
+      controller.instance_variable_set(:@edit,
+                                       :new      => {:flavor => flavor.id},
+                                       :explorer => true)
+      expect_any_instance_of(VmCloud).to receive(:resize).with(flavor)
+      post :resize_vm, :params => {
+        :button => 'submit',
+        :id     => vm_openstack.id
+      }
+      expect(response.status).to eq(200)
```

Comment:
@h-kataria  I'm not sure what expectation I can write into the test to verify it was actually resized. There doesn't appear to be any state that would change on the mock that the test would be able to catch. (looking at `app/models/manageiq/providers/openstack/cloud_manager/vm/resize.rb:raw_resize`)

Aside from that, I would assume that actually testing that the resize functionality actually does something once `resize` is called is outside the scope of the UI test?

Happy to have you correct my misunderstandings here in either case :)